specification/sbi_atee.adoc

```diff
@@ -10,8 +39,1358 @@ shown in the figure above, the TEE Runtime (operating in VS-mode) supports
+     * page if applicable), and then calling sbi_tee_host_run_tvm_vcpu to resume execution at
+     * the following instruction. 
+    */
+    GUEST_LOAD_PAGE_FAULT = 21,
+    /* 
+     * The TVM executed an instruction that caused an exit. The host should decode the
+     * instruction by examining `stval` from `tvm_vcpu_supervisor_csrs`, and determine
+     * the further course of action, and calling then calling sbi_tee_host_run_tvm_vcpu
+     * if appropriate to resume execution at the following instruction.
+    */
+    VIRTUAL_INSTRUCTION = 22,
+    /* 
+     * The TVM encounted a store fault in a confidential, MMIO, or shared-memory
+     * region. The host should determine the fault address by retrieving the
+     * `htval` from `tvm_vcpu_hypervisor_csrs` and `stval` from `tvm_vcpu_supervisor_csrs`
+     * and combining them as follows: "(htval << 2) | (stval & 0x3)". The fault address
+     * can then be used to determine the type of memory region, and making the appropriate
+     * call (example: sbi_tee_host_add_tvm_zero_pages() to add a demand-zero confidential
+     * page if applicable), and then calling sbi_tee_host_run_tvm_vcpu to resume execution at
+     * the following instruction.
+     */
+    GUEST_STORE_PAGE_FAULT = 23,
+};
+-------
+
+== TEE Interrupt Extension (EID #0x54414949)
+The TEE Interrupt extension supplements the TEE Host extension with hardware-assisted interrupt
+virtualization using the RISC-V Advanced Interrupt Architecture (AIA) on platforms which
+support it.
+
+=== Function: TEE Interrupt Init TVM AIA (FID #0)
+[source, C]
+-------
+struct sbiret sbi_tee_interrupt_init_tvm_aia(unsigned long tvm_guest_id,
+                                             unsigned long tvm_aia_params_addr,
+                                             unsigned long tvm_aia_params_len);
+-------
+
+Configures AIA virtualization for the TVM identified by `tvm_guest_id` based on the
+parameters in the `tvm_aia_params` structure at the non-confidential physical address
+at `tvm_aia_params_addr`. The `tvm_aia_params_len` is the byte-length of the `tvm_aia_params` 
+structure.
+
+This cannot be called after `sbi_tee_host_finalize_tvm()`.
+
+The format and semantics of the `tvm_aia_params_addr` structure appears below.
+
+[source, C]
+-------
+struct tvm_aia_params {
+    /* 
+     * The base address of the virtualized IMSIC in TVM physical address space.
+     *
+     * IMSIC addresses follow the below pattern:
+     *
+     * XLEN-1 >=24 12 0 | | | |
+     *
+     * |xxxxxx|Group Index|xxxxxxxxxxx|Hart Index|Guest Index| 0 |
+     *
+     * The base address is the address of the IMSIC with group ID, hart ID, and guest ID of 0.
+     */
+    unsigned long imsic_base_addr;
+    /* The number of group index bits in an IMSIC address. */
+    unsigned int group_index_bits;
+    /* The location of the group index in an IMSIC address. Must be >= 24. */
+    unsigned int group_index_shift;
```

Comment:
General comment (and maybe @atishp04 has an opinion here too): I feel like we should try to use explicitly-typed sizes in in-memory structures that cross the ABI boundary, with a possible exception of using `unsigned long` for pointer types (though 64-bit would still be fine for RV32 since physical addresses are 34-bits there).

Reply:
Changed to uint32_t to match the previous uint16_t from above, but will defer to @atishp04 if unsigned int / unsigned short is preferred.

Reply:
yeah. That is better. I already tripped once on this ;)